### ISSUE TITLE
BUG: qsv frequency -l ignores limits greater than 10

### ISSUE THREAD

**Issue by tmtmtmtm** [jqnatividad/qsv#1962]
given the following CSV file:

```sh
(for i in $(seq 2001 2024); do for j in $(seq 1 $((i-2000))); do echo "$i"; done; done) | qsv rename -n year > data.csv
```

then:

* `qsv frequency data.csv` correctly gives the 10 most recent years.
* `qsv frequency -l 0 data.csv` correctly counts every year
* `qsv frequency -l 3 data.csv` (or any number up to 10) correctly truncates to that number
* `qsv frequency -l 20 data.csv` (or any number greater than 10) only gives 10 results

This is with qsv 0.128.0 (several different builds). On 0.123.0 it correctly returns 20, so there's a regression sometime after that, but I don't have any other versions installed atm to track it down further.

**Comment by jqnatividad** [comment]
Thanks for the report @tmtmtmtm .

I'm currently refactoring `frequency` to make it cardinality-aware (#1934 ), so I'll fix it while doing so.

**Comment by jqnatividad** [comment]
As it turns out, this bug is caused by `frequency` incorrectly detecting a column with all unique identifiers, so #1934 to make use of the stats cache will fix this as well as we're fetching a column's cardinality ahead of time.

**Comment by jqnatividad** [comment]
Fixed this before finishing #1934 so 0.129.0 can be released.
